homeassistant/components/azure_event_hub/__init__.py

```diff
@@ -45,54 +42,77 @@
     {
         DOMAIN: vol.Schema(
             {
-                vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
-                vol.Exclusive(CONF_EVENT_HUB_CON_STRING, "setup_methods"): cv.string,
-                vol.Exclusive(CONF_EVENT_HUB_NAMESPACE, "setup_methods"): cv.string,
+                vol.Optional(CONF_EVENT_HUB_INSTANCE_NAME): cv.string,
+                vol.Optional(CONF_EVENT_HUB_CON_STRING): cv.string,
+                vol.Optional(CONF_EVENT_HUB_NAMESPACE): cv.string,
                 vol.Optional(CONF_EVENT_HUB_SAS_POLICY): cv.string,
                 vol.Optional(CONF_EVENT_HUB_SAS_KEY): cv.string,
-                vol.Optional(CONF_SEND_INTERVAL, default=5): cv.positive_int,
-                vol.Optional(CONF_MAX_DELAY, default=30): cv.positive_int,
+                vol.Optional(CONF_SEND_INTERVAL): cv.positive_int,
+                vol.Optional(CONF_MAX_DELAY): cv.positive_int,
                 vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
             },
-            cv.has_at_least_one_key(
-                CONF_EVENT_HUB_CON_STRING, CONF_EVENT_HUB_NAMESPACE
-            ),
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
 async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
-    """Activate Azure EH component."""
-    config = yaml_config[DOMAIN]
-    if config.get(CONF_EVENT_HUB_CON_STRING):
-        client_args = {
-            "conn_str": config[CONF_EVENT_HUB_CON_STRING],
-            "eventhub_name": config[CONF_EVENT_HUB_INSTANCE_NAME],
-        }
-        conn_str_client = True
-    else:
-        client_args = {
-            "fully_qualified_namespace": f"{config[CONF_EVENT_HUB_NAMESPACE]}.servicebus.windows.net",
-            "eventhub_name": config[CONF_EVENT_HUB_INSTANCE_NAME],
-            "credential": EventHubSharedKeyCredential(
-                policy=config[CONF_EVENT_HUB_SAS_POLICY],
-                key=config[CONF_EVENT_HUB_SAS_KEY],
-            ),
-        }
-        conn_str_client = False
-
-    instance = hass.data[DOMAIN] = AzureEventHub(
+    """Activate Azure EH component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit deprecated warning and pass the rest to the config flow.
```

Comment:
```suggestion
    Adds an empty filter to hass data.
    Tries to get a filter from yaml, if present set to hass data.
    If config is empty after getting the filter, return,
    otherwise emit deprecated warning
    and pass the rest to the config flow.
```